src/cargo/util/toml/schema.rs

```diff
@@ -0,0 +1,1134 @@
+    #[serde(skip)]
+    pub lints: Option<TomlLints>,
+
+    pub version: Option<semver::Version>,
+    pub authors: Option<Vec<String>>,
+    pub description: Option<String>,
+    pub homepage: Option<String>,
+    pub documentation: Option<String>,
+    pub readme: Option<StringOrBool>,
+    pub keywords: Option<Vec<String>>,
+    pub categories: Option<Vec<String>>,
+    pub license: Option<String>,
+    pub license_file: Option<String>,
+    pub repository: Option<String>,
+    pub publish: Option<VecStringOrBool>,
+    pub edition: Option<String>,
+    pub badges: Option<BTreeMap<String, BTreeMap<String, String>>>,
+    pub exclude: Option<Vec<String>>,
+    pub include: Option<Vec<String>>,
+    pub rust_version: Option<RustVersion>,
+    // We use skip here since it will never be present when deserializing
+    // and we don't want it present when serializing
+    #[serde(skip)]
+    pub ws_root: PathBuf,
+}
+
+/// Represents the `package`/`project` sections of a `Cargo.toml`.
+///
+/// Note that the order of the fields matters, since this is the order they
+/// are serialized to a TOML file. For example, you cannot have values after
+/// the field `metadata`, since it is a table and values cannot appear after
+/// tables.
+#[derive(Deserialize, Serialize, Clone, Debug)]
+#[serde(rename_all = "kebab-case")]
+pub struct TomlPackage {
+    pub edition: Option<MaybeWorkspaceString>,
+    pub rust_version: Option<MaybeWorkspaceRustVersion>,
+    pub name: String,
+    pub version: Option<MaybeWorkspaceSemverVersion>,
+    pub authors: Option<MaybeWorkspaceVecString>,
+    pub build: Option<StringOrBool>,
+    pub metabuild: Option<StringOrVec>,
+    pub default_target: Option<String>,
+    pub forced_target: Option<String>,
+    pub links: Option<String>,
+    pub exclude: Option<MaybeWorkspaceVecString>,
+    pub include: Option<MaybeWorkspaceVecString>,
+    pub publish: Option<MaybeWorkspaceVecStringOrBool>,
+    pub workspace: Option<String>,
+    pub im_a_teapot: Option<bool>,
+    pub autobins: Option<bool>,
+    pub autoexamples: Option<bool>,
+    pub autotests: Option<bool>,
+    pub autobenches: Option<bool>,
+    pub default_run: Option<String>,
+
+    // Package metadata.
+    pub description: Option<MaybeWorkspaceString>,
+    pub homepage: Option<MaybeWorkspaceString>,
+    pub documentation: Option<MaybeWorkspaceString>,
+    pub readme: Option<MaybeWorkspaceStringOrBool>,
+    pub keywords: Option<MaybeWorkspaceVecString>,
+    pub categories: Option<MaybeWorkspaceVecString>,
+    pub license: Option<MaybeWorkspaceString>,
+    pub license_file: Option<MaybeWorkspaceString>,
+    pub repository: Option<MaybeWorkspaceString>,
+    pub resolver: Option<String>,
+
+    pub metadata: Option<toml::Value>,
+
+    /// Provide a helpful error message for a common user error.
+    #[serde(rename = "cargo-features", skip_serializing)]
+    pub _invalid_cargo_features: Option<InvalidCargoFeatures>,
```

Comment:
What is the use case of making this public?

Reply:
That depends on if we want this to be `#[non_exhaustive]` or directly constructable.  For now, I was going with directly constructable.